lib/std/debug.zig

```diff
@@ -106,6 +106,63 @@ pub fn getSelfDebugInfo() !*DebugInfo {
     }
 }
 
+/// Tries to print a hexadecimal view of the bytes, unbuffered, and ignores any error returned.
+pub fn hexdump(bytes: []const u8) void {
+    hexdump_internal(bytes) catch {};
+}
+
+fn hexdump_internal(bytes: []const u8) !void {
```

Comment:
Might be nice to make this public and make it take a writer + `TTY.Config` for anyone that wants to use a different combination of writer/TTY.

Maybe something like
```suggestion
pub fn hexdumpAdvanced(bytes: []const u8, writer: anytype, ttyconf: std.io.tty.Config) !void {
```

(the `Advanced` suffix is borrowed from some `std.mem.Allocator` function names; I'm sure there's a better name than `hexdumpAdvanced` though)

Reply:
Hmm and when would this be useful? At this point the code could almost be used for hexdumps in actual applications (maybe a programming puzzle game or something). Is this what we want? I feel like at this point people would probably want to write the code themselves if they want more than this.

Maybe a use case would indeed be to write the hexdump to something other than stderr in the terminal? Like a file on disk? Or maybe to a buffer of text that is drawn to the screen through a graphics API? Is something like that what you had in mind?

Reply:
> Or maybe to a buffer of text that is drawn to the screen through a graphics API?

I can see how this might be useful for debugging but currently I think basically all `std.debug` functions just use stderr by default with no customizability. I almost feel like this is a more general problem that should be solved for the other functions too.

Reply:
`std.io.tty.Config` is currently only used to make the addresses at the start dimmed. If at some point there is someone that has a specific use case for not wanting to dim the addresses of some debugging function then they can make a PR and I would be interested to hear the use case.
If that happens I think it should be a second function (like `std.debug.hexdumpAdvanced`) so that the main `std.debug.hexdump` stays short to type and only takes one argument and returns no error for quick debugging.

Reply:
Just to be clear, I'm not suggesting changing `std.debug.hexdump`, I'm suggesting changing the implementations to:

```zig
pub fn hexdump(bytes: []const u8) void {
    const stderr = std.io.getStdErr();
    const ttyconf = std.io.tty.detectConfig(stderr);
    const writer = stderr.writer();
    hexdumpAdvanced(bytes, writer, ttyconf) catch {};
}

pub fn hexdumpAdvanced(bytes: []const u8, writer: anytype, ttyconf: std.io.TTY.Config) !void {
    var chunks = mem.window(u8, bytes, 16, 16);
    // ...
}
```

Reply:
Yes, that is how I understood it as well but as I said, I don't see the use case for this right now so I'm suggesting holding off of anymore changes until someone actually needs this. I named some possible use cases above but it goes back to that there should probably be a more general solution to this problem.